app/src/main/java/hr/dice/filipbionda/tmdbpractice/ui/homescreen/HomeScreen.kt

```diff
@@ -0,0 +1,558 @@
+            FilterChip(
+                modifier = Modifier
+                    .padding(
+                        start =
+                        if (category == initialCategory || category == categories.last()) {
+                            dimensionResource(
+                                R.dimen.home_screen_horizontal_padding,
+                            )
+                        } else {
+                            dimensionResource(R.dimen.filter_chip_group_content_padding)
+                        },
+                    )
+                    .height(dimensionResource(R.dimen.filter_chip_height))
+                    .width(dimensionResource(R.dimen.filter_chip_width)),
+                selected = selectedChip == category,
+                shape = RoundedCornerShape(dimensionResource(R.dimen.filter_chip_shape_size)),
+                border = null,
+                colors = FilterChipDefaults.filterChipColors(
+                    selectedContainerColor = MaterialTheme.colorScheme.secondary,
+                    selectedLabelColor = MaterialTheme.colorScheme.secondary,
+                    disabledLabelColor = MaterialTheme.colorScheme.onSurface,
+                    disabledSelectedContainerColor = MaterialTheme.colorScheme.onBackground,
+                ),
+                onClick = {
+                    selectedChip = category
+                    onChipClick(selectedChip)
+                },
+                label = {
+                    Text(
+                        text = createChipLabel(category),
+                        style = if (category == selectedChip) {
+                            MaterialTheme.typography.labelSmall
+                        } else {
+                            MaterialTheme.typography.labelSmall.copy(
+                                color = grey_50,
+                                fontWeight = FontWeight.Normal,
+                            )
+                        },
+                        textAlign = TextAlign.Center,
+                        modifier = Modifier.fillMaxWidth(),
+                    )
+                },
+            )
+        }
+    }
+}
+
+@Composable
+fun HomeScreenContent(
+    items: List<String>,
+    popularItems: List<String>,
+    modifier: Modifier = Modifier,
+) {
+    val mockImage =
+        when (items.first()) {
+            "movie1" -> mockMovieUrl
+            "series1" -> mockSeriesUrl
+            "anime1" -> mockAnimeUrl
+            "soaps1" -> mockSoapsUrl
+            "specials1" -> mockSpecialsUrl
+            else -> throw IllegalArgumentException()
+        }
+    val itemsLazyRowState = rememberLazyListState()
+    val popularItemsLazyRowState = rememberLazyListState()
+
+    LaunchedEffect(items, popularItems) {
+        launch {
+            listOf(
+                launch { itemsLazyRowState.animateScrollToItem(0) },
+                launch { popularItemsLazyRowState.animateScrollToItem(0) },
+            ).forEach { it.join() }
```

Comment:
Why do you need the join?

Reply:
It works without listOf() and .forEache { it.join() }, thanks!